bin/studio-common

```diff
@@ -17,6 +17,59 @@
 # limitations under the License.
 #
 
+# This value is automatically updated by Expeditor to reflect .hab-version
+export SUPPORTED_HAB_VERSION="0.34.1"
+
+# Ensure that the installed version of the hab toolchain is >= to the required version.
+#
+# Given the rapid development and release cadence of Habitat, we want to ensure that
+# developers are using a version of Habitat is is well tested against both this library
+# and the supporting automation (i.e. Expeditor builds).
+required_version="$SUPPORTED_HAB_VERSION"
+installed_version=$(hab -V | awk '{print $NF}' | cut -f1 -d/);
+
+MAJOR=$(echo "$installed_version"  | cut -f1 -d.);
+MINOR=$(echo "$installed_version"  | cut -f2 -d.);
+PATCH=$(echo "$installed_version"  | cut -f3 -d.);
+R_MAJOR=$(echo "$required_version" | cut -f1 -d.);
+R_MINOR=$(echo "$required_version" | cut -f2 -d.);
+R_PATCH=$(echo "$required_version" | cut -f3 -d.);
+
+need_upgrade=false;
+is_newer=false;
+
+[[ $PATCH -lt $R_PATCH ]] && need_upgrade=true;
+if [[ $MINOR -ne $R_MINOR ]]; then
+  [[ $MINOR -lt $R_MINOR ]] && need_upgrade=true || need_upgrade=false;
+  [[ $MINOR -gt $R_MINOR ]] && is_newer=true || is_newer=false;
+fi;
+if [[ $MAJOR -ne $R_MAJOR ]]; then
+  [[ $MAJOR -lt $R_MAJOR ]] && need_upgrade=true || need_upgrade=false;
+  [[ $MAJOR -gt $R_MAJOR ]] && is_newer=true || is_newer=false;
+fi;
+
+if $need_upgrade; then
+  print_version_error_message && exit 1;
```

Comment:
don't you need the functions to be defined first?

Reply:
yes, I'll reorder.